src/liboslcomp/ast.cpp

```diff
@@ -76,6 +77,41 @@ ScopeExit print_node_counts ([](){
 
 
 
+ASTNode *
+reverse (ASTNode *list)
+{
+    // trivial case: empty or just one item
+    if (!list || !list->nextptr())
+        return list;
+
+#ifndef OIIO_REFCNT_HAS_RELEASE
+    ASSERT (0 && "Do not call reverse() if you didn't build with an OIIO "
+                 "new enough to have intrustive_ptr::release().");
+    return nullptr;
+#else
+    // Turn the list (whose `next` fields are ref-counted pointers) into
```

Comment:
Why can't you do something like this?

```
ASTNode::ref reverse(ASTNode::ref root) {
  ASTNode::ref new_root = nullptr;
  while (root) {
    ASTNode::ref next = root->next;
    root->next = new_root;
    new_root = root;
    root = next;
  }
  return new_root;
}
```

Why do you need to allocate a whole new vector + do 3 passes?

Reply:
You're right, this is simpler.